File: okex-php-sdk-api-v5/src/ConvertApi.php
<?php


namespace okv5;


class ConvertAPI extends Utils
{

    public function currencies(){
        $params = [
        ];

        return $this->request('/api/v5/asset/convert/currencies', $params, 'GET');
    }

    public function currencyPair($fromCc,$toCcy){
        $params = [
            'fromCcy' => $fromCcy,
            'toCcy' => $toCcy,
        ];

        return $this->request('/api/v5/asset/convert/currency-pair', $params, 'GET');
    }

    public function estimateQuote($baseCcy,$quoteCcy,$side,$rfqSz,$rfqSzCcy,$clQReqId='',$tag=''){
        $params = [
            'baseCcy' => $baseCcy,
            'quoteCcy' => $quoteCcy,
            'side' => $quotesideCcy,
            'rfqSz' => $rfqSz,
            'rfqSzCcy' => $rfqSzCcy,
            'clQReqId' => $clQReqId,
            'tag' => $tag,
        ];

        return $this->request('/api/v5/asset/convert/estimate-quote', $params, 'POST');
    }

    public function trade($quoteId,$baseCcy,$quoteCcy,$side,$sz,$szCcy,$clTReqId='',$tag=''){
        $params = [
            'quoteId' => $quoteId,
            'baseCcy' => $baseCcy,
            'quoteCcy' => $quoteCcy,
            'side' => $quotesideCcy,
            'sz' => $sz,
            'szCcy' => $szCcy,
            'clTReqId' => $clTReqId,
            'tag' => $tag,
        ];

        return $this->request('/api/v5/asset/convert/trade', $params, 'POST');
    }

    public function history($after='',$before='',$limit='',$tag=''){
        $params = [
            'after' => $after,
            'before' => $before,
            'limit' => $limit,
            'tag' => $tag,
        ];

        return $this->request('/api/v5/asset/convert/history', $params, 'GET');
    }

    
}

File: okex-java-sdk-api-v5/src/main/java/com/okex/open/api/service/convert/impl/ConvertAPI.java
package com.okex.open.api.service.convert.impl;

import com.alibaba.fastjson.JSONObject;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Query;

public interface ConvertAPI {

    //获取闪兑币种列表 Get convert currencies
    @GET("/api/v5/asset/convert/currencies")
    Call<JSONObject> getCurrencies();

    //获取闪兑币对信息 Get convert currency pair
    @GET("/api/v5/asset/convert/currency-pair")
    Call<JSONObject> getCurrencyPair(@Query("fromCcy") String fromCcy,
                                     @Query("toCcy") String toCcy);

    //闪兑预估询价 Estimate quote
    @POST("/api/v5/asset/convert/estimate-quote")
    Call<JSONObject> estimateQuote(@Body JSONObject jsonObject);

    //闪兑交易 Convert trade
    @POST("/api/v5/asset/convert/trade")
    Call<JSONObject> trade(@Body JSONObject jsonObject);

    //获取闪兑交易历史 Get convert history
    @GET("/api/v5/asset/convert/history")
    Call<JSONObject> getHistory(@Query("clTReqId") String clTReqId,
                                @Query("after") String after,
                                @Query("before") String before,
                                @Query("limit") String limit,
                                @Query("tag") String tag);

}
